### PR TITLE
Install golangci-lint in the local tools dir

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
Golangci-lint output can vary significantly between versions, and each project uses a different one. It makes for a better local dev experience if we explicitly install the version we want, just as we do with other local tools.

I've also made the CI use this for consistency's sake.

### CODE REVIEW

**Review by pavolloffay — APPROVED** [review]

File: .github/workflows/continuous-integration.yaml [hunk 1]
```diff
@@ -29,9 +29,6 @@ jobs:
   lint:
     name: Code standards (linting)
     runs-on: ubuntu-20.04
-    strategy:
-      matrix:
-        workdir: [".", "./cmd/otel-allocator", "./cmd/operator-opamp-bridge"]
```
Comment (pavolloffay) [inline .github/workflows/continuous-integration.yaml:34]: why was this removed? 

IIRC this was assuming the lint runs per all subprojects

Comment (swiatekm-sumo) [reply .github/workflows/continuous-integration.yaml:34]: The lint target in our Makefile already does this, so it seems redundant unless we want to have each subproject check show up as a separate CI job.

**Comment by pavolloffay** [comment]
created https://github.com/open-telemetry/community/issues/1579 to unblock merging.

**Review by jaronoff97 — APPROVED** [review]
